src/DomainLib.Projections.Sql/ISqlDialect.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.Data.Common;
+
+namespace DomainLib.Projections.Sql
+{
+    public interface ISqlDialect
+    {
```

Comment:
I thought about that too. The interface has since been renamed to IDbConnector as it's more than just a SQL dialect now.
I did wonder about separating out the part that generates SQL from the part that provides the connection/binds parameters.

Those two things are coupled in some way due to the fact that they're both related to the database type. Agree that it's not full SRP at the moment though. We can think some more about it later

Reply:
I've removed some of the uses of DbCommand now, as I've changed the DbConnector to return strings for the upsert and delete commands.
That allows us to consolidate the SQL that's run if a user wants to do an upsert/delete and run some custom sql for a projection.
The rest are going to have to stay as it's only the abstract classes that provide async overloads when talking to the db